cronet/src/main/java/io/grpc/cronet/CronetChannelBuilder.java

```diff
@@ -271,12 +276,50 @@ public void close() {
           ((ExperimentalCronetEngine) cronetEngine)
               .newBidirectionalStreamBuilder(url, callback, executor);
       if (trafficStatsTagSet) {
-        builder.setTrafficStatsTag(trafficStatsTag);
+        setTrafficStatsTag(builder, trafficStatsTag);
       }
       if (trafficStatsUidSet) {
-        builder.setTrafficStatsUid(trafficStatsUid);
+        setTrafficStatsUid(builder, trafficStatsUid);
       }
       return builder;
     }
+
+    private static void setTrafficStatsTag(ExperimentalBidirectionalStream.Builder builder,
+        int tag) {
+      Exception caughtException = null;
+      try {
+        Method setTagMethod = ExperimentalBidirectionalStream.Builder.class
```

Comment:
The getMethod() is the slowest part of reflection. We should save the result and re-use it across invoke()s.

Since we really aren't expecting this to ever fail (because the app will be shipping cronet itself), then performance for the failure case probably doesn't matter much. That means we could just re-call getMethod() each time if it is failing, and so we'd have an exception each time as well. We'd only need a single field to hold each Method.

We can use a lock or volatile to protect the Method field. It doesn't matter to me which.

Reply:
Done.